### DISCUSSION TITLE
Problem of CHF calculation in two phase flow

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27520] [Q&A Modules: Thermal Hydraulics] [answered]
Hi @GiudGiud and @joshuahansel ,

I have developed a two-phase flow code (six-equation model) according to THM. I also use [FlowChannel2Phase], [HeatStructureCylindrical], and [HeatTransferFromHeatStructure2Phase] components, just like THM does.

Now I am doing the CHF calculation, but some unexpected oscillation occurred.
![image](https://github.com/idaholab/moose/assets/118275988/b0df8022-6f08-4a14-8ee9-75482029c158)

The T_wall in the solid region is [FIRST, LAGRANGE], but all variables in the 1-D fluid region are [CONSTANT, MONOMIAL]. I use `ProjectionAux` to show T_wall in elemental form. It may be correct. 
![image](https://github.com/idaholab/moose/assets/118275988/44367070-df2c-410d-94b4-f0d94b5ca36e)

But the T_wall in the solid region is still oscillating. Do you know the reason why the T_wall is oscillating? And how to fix this problem?

**Comment by GiudGiud** [comment]
Hello

Did you try to refine the mesh?
Should you have some sort of limiters to capture that gradient in temperature more smoothly?
I imagine temperature is not the only variable with such a profile

Guillaume

**Comment by joshuahansel** [comment]
The CFEM discretization in the heat structures is not immune to oscillations; if you have a sharp jump in the boundary conditions, this is possible. I don't think the `ProjectionAux` helps you, as it just masks the problem. The quick fixes to *reduce* the problem are refining the mesh and using a smoother transition in the boundary condition. The full fix would be some kind of change to the sptail scheme in the heat structure, like a stabilization of some kind, which I admit I haven't investigated yet (the oscillations I've encountered have not been so bad).

**Reply by SomnusYu** [reply]
Hi @joshuahansel and @GiudGiud,

I tried to refine the mesh and use MINMOD slop limiters, but the result still has oscillation.

![image](https://github.com/idaholab/moose/assets/118275988/ed417b22-abc6-49a6-8d4b-96a273817ea9)

> using a smoother transition in the boundary condition

The boundary condition in the solid region is the sum of qw_liq and qw_vap. The qw_vap is indeed oscillating. I'm not sure which method should I use to smooth the boundary condition (smooth qw_vap). Could I apply the method in `ProjectionAux` to the ADmaterial?

**Reply by joshuahansel** [reply]
By smoothing the BC, I was thinking more of just making a gradual transition from your BC on the left to the BC on the right.

For your question about `ProjectionAux`, I'm not sure if you could use that somehow and if it would help or not.

**Reply by SomnusYu** [reply]
> I was thinking more of just making a gradual transition from your BC on the left to the BC on the right.

Do you have any suggestions on how to smooth the BC?

**Reply by joshuahansel** [reply, marked as answer]
For example, you can gradually transition your heat transfer coefficient using something like `SmoothTransitionFunction`.

**Reply by SomnusYu** [reply]
OK, I'll have a try. Thanks!
